### PR TITLE
Set collator minimum bond to 4M KMA

### LINKED ISSUE #933: Set collator limit to 4M KMA
As announced, the temporary staking program will be ending with v4.0.0

Collators shall now only be selected for block production if they have a self-bond of 4M KMA or higher.

### PR DESCRIPTION
**Pull request by Garandor**
## Description

Calamari Functional change:
- collators below 4mil are no longer considered for collator assignment
https://github.com/Manta-Network/Manta/pull/937/files#diff-b616758683d3c1926eb58a1b16a2b439122f36e4738af354b42addc5daf2cc83R22

Testing changes:
- add regression test for 400k-bonded collators
- add `parachain_staking` to integration_test mock
- refactored integration test for `lazy_static` ALICE-FERDIE keys

relates to OR closes: #933

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] This PR is targeted against the *current*  Milestone ( otherwise discuss if it can be added in time)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

File: runtime/calamari/tests/integrations_mock/integration_tests.rs [hunk 1]
```diff
@@ -581,49 +660,44 @@ fn sanity_check_round_duration() {
...
+        ])
+        .build()
+        .execute_with(|| {
+            // Create and bond session keys to Bob's account.
+            assert_ok!(Session::set_keys(
+                Origin::signed(BOB.clone()),
+                BOB_SESSION_KEYS.clone(),
+                vec![]
+            ));
+            // assert_eq!(
```
Comment (ghzlatarev) [inline runtime/calamari/tests/integrations_mock/integration_tests.rs]: Commented code.

**Review by ghzlatarev — COMMENTED** [review]
Looks good, linters are failing though.
Also are there some easy tests that can be moved from teh QA sheet ?

File: runtime/calamari/tests/integrations_mock/integration_tests.rs [hunk 2]
```diff
@@ -570,6 +547,108 @@ fn sanity_check_round_duration() {
...
+            (BOB.clone(), BOB_SESSION_KEYS.clone()),
+            (CHARLIE.clone(), CHARLIE_SESSION_KEYS.clone()),
+            (DAVE.clone(), DAVE_SESSION_KEYS.clone()),
+            (EVE.clone(), EVE_SESSION_KEYS.clone()),
+            (FERDIE.clone(), FERDIE_SESSION_KEYS.clone()),
+        ])
+        .build()
+        .execute_with(|| {
+            // 0. initialize collators as candidates with manta_collator_selection
+            assert_ok!(CollatorSelection::set_desired_candidates(root_origin(), 6));
```
Comment (Dengjianping) [inline runtime/calamari/tests/integrations_mock/integration_tests.rs]:
```rust
let candidates_count = 6;
```

File: runtime/calamari/tests/integrations_mock/mod.rs [hunk 3]
```diff
@@ -19,14 +19,42 @@
...
-use manta_primitives::types::Balance;
+use lazy_static::lazy_static;
+use manta_primitives::types::{AccountId, Balance};
+use session_key_primitives::util::{unchecked_account_id, unchecked_collator_keys};
+use sp_core::sr25519::Public;
+#[cfg(feature = "std")]
+pub(crate) use std::clone::Clone;
 
-pub const BOND_AMOUNT: Balance = 1_000 * KMA;
+pub const COLLATOR_MIN_BOND: Balance = 4_000_000 * KMA;
```
Comment (Dengjianping) [inline runtime/calamari/tests/integrations_mock/mod.rs]: Seems both bond values can be imported from `staking.rs`.

File: runtime/calamari/tests/integrations_mock/mod.rs [hunk 4]
```diff
@@ -19,14 +19,42 @@
...
 pub use calamari_runtime::{currency::KMA, Event, Origin, Runtime, System};
 
 use frame_support::weights::{DispatchInfo, Weight};
-use manta_primitives::types::Balance;
+use lazy_static::lazy_static;
+use manta_primitives::types::{AccountId, Balance};
+use session_key_primitives::util::{unchecked_account_id, unchecked_collator_keys};
+use sp_core::sr25519::Public;
+#[cfg(feature = "std")]
+pub(crate) use std::clone::Clone;
```
Comment (Dengjianping) [inline runtime/calamari/tests/integrations_mock/mod.rs]: Not sure why `Clone` is imported, it should be in `std::prelude`.

**Review by ferrell-code — DISMISSED** [review]

File: runtime/calamari/tests/integrations_mock/mod.rs (hunk 4 above)
Comment (Garandor) [reply runtime/calamari/tests/integrations_mock/mod.rs]: Has to do with `lazy_static!`
https://github.com/Manta-Network/Manta/blob/4a659e980400e2a267b6ee00f69b17b2767224cb/runtime/calamari/tests/integrations_mock/mod.rs#L39
The type it creates implements `Clone`, but the other sources for some reason don't have access to the `clone()` method unless `std::Clone` is explicitly `use`d.
Not sure why tho

File: runtime/calamari/tests/integrations_mock/mod.rs (hunk 3 above)
Comment (Garandor) [reply runtime/calamari/tests/integrations_mock/mod.rs]: good call, thanks!

File: runtime/calamari/tests/integrations_mock/integration_tests.rs (hunk 2 above)
Comment (Garandor) [reply runtime/calamari/tests/integrations_mock/integration_tests.rs]: refactored it out, this is now moot

File: runtime/calamari/tests/integrations_mock/mod.rs (hunk 4 above)
Comment (Garandor) [reply runtime/calamari/tests/integrations_mock/mod.rs]: Upgrading compiler fixed this, i removed the explicit import

**Comment by Garandor** [comment]
> Also are there some easy tests that can be moved from teh QA sheet ?

Addressed all comments except the one above, i'll leave this to another PR

**Review by ghzlatarev — APPROVED** [review]

**Review by ferrell-code — APPROVED** [review]

**Pull request merged**